openapi-bal-service/src/main/java/io/ballerina/openapi/converter/Constants.java

```diff
@@ -82,6 +82,7 @@ public class Constants {
     public static final String WILD_CARD_SUMMARY = "Any type of entity body";
     public static final String MEDIA_TYPE = "mediaType";
     public static final String TUPLE = "tuple";
+    public static final String REGEX_INTERPOLATION_PATTERN = "^(?!.*\\$\\{).+$";
```

Comment:
We need to update this regex to include literal `$` character once the https://github.com/ballerina-platform/ballerina-lang/issues/41492 is fixed.
So Lets create an issue in the open API repo describing the above scenario and add the issue link as a comment in the above lang issue.

Reply:
Shall we add the negative test for these scenarios to check the compilation issue? once the lang fix has been done, our tool side will notice it. :)

Reply:
I have added an issue in standard library repo about this. Since Ballerina gives us compiler errors when using REGEX patterns which includes `\${1,2}` or `\${2}`, we skip all the patterns which includes `${` and consider them as invalid patterns or patterns which contains interpolation when mapping Ballerina constraints to OpenAPI specification.
Once [ballerina-platform/ballerina-lang/issues/41492](https://github.com/ballerina-platform/ballerina-lang/issues/41492) is fixed, we need to improve [ballerina-platform/ballerina-standard-library/issues/5048](https://github.com/ballerina-platform/ballerina-standard-library/issues/5048) this as well.
I'll add negative test cases to check compilation issues as well..

Reply:
@lnash94 Shall I add these negative tests in a separate PR where I'm going to address all the negative cases via that PR WDYT?